File: Python/백준/Bronze/9063. 대지/대지.py
import sys

n = int(sys.stdin.readline().rstrip())
x_list, y_list = [], []

for i in range(n):
    x, y = map(int, sys.stdin.readline().split())
    x_list.append(x)
    y_list.append(y)

min_x, max_x = min(x_list), max(x_list)
min_y, max_y = min(y_list), max(y_list)

print((max_x - min_x) * (max_y - min_y))

File: C++17/백준/Bronze/9063. 대지/대지.cc
#include <iostream>
#define MAX 10000
#define MIN (-10000)

using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(0);

    int n;
    cin >> n;

    int minX = MAX;
    int maxX = MIN;
    int minY = MAX;
    int maxY = MIN;

    for (int i = 0; i < n; i++) {
        int x, y;
        cin >> x >> y;

        if (x < minX) {
            minX = x;
        }

        if (x > maxX) {
            maxX = x;
        }

        if (y < minY) {
            minY = y;
        }

        if (y > maxY) {
            maxY = y;
        }
    }

    int area = (maxX - minX) * (maxY - minY);

    cout << (area < 0 ? 0 : area) << "\n";

    return 0;
}
